### ISSUE TITLE
[Feature]: Keyboard navigation of table inline editing

### ISSUE THREAD

**Issue by adrian-stephens** [appsmithorg/appsmith#23961]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

**Tab doesn't do anything useful when editing a row in a table widget - either a new row, or an existing row.
I think users are used to tab moving to the next field on the right in a similar situation in a spreadsheet.

So I am requesting that the tab key complete editing of the current field,  set the focus to the next field to the right
and enable editing of it.  A double tab ("tab through") should leave the cell skipped over unchanged.

This is a particularly convincing need in the context of "new row",  when it is expected that many of the fields in the row
will need to be specified.

Test system:  Brave or Firefox browser on linux, latest (at the time of writing) version of appsmith, self hosted. 

### Why should this be worked on?

It will improve user experience, and reduce frustration and surprise when the tab doesn't act "as expected".

**Comment by lk-smit-ag** [comment]
Is there any update on this?
This feels really strange to edit data fastly...

**Comment by expensiveeyeballs** [comment]
Hmmm...this thread started in 2023 and still there's no tab navigation in tables? I just emailed support about this and they directed me here and requested that I specify my use case. So I guess I'll add my two cents even-though it seems Appsmith has made it pretty clear that they're not interested in including this very basic core functionality. Of course people expect this feature when doing data entry and it's very very strange not to have this. Use case is: uh, look at every other product in the world and see how it works. But also, their table widget seems so unstable in so many other ways - intermittent data binding failure on select type columns, property toggles with more than two states or that inadvertently break other un-related features - that the entire table widget just needs an overhaul.

**Comment by sjr267** [comment]
This feature appeared a few months ago and disappeared earlier this month, possibly at v1.88. One of the fixes in v1.88 was for tab navigation issues.
